### DISCUSSION TITLE
Pimcore 7 - PHP rendering engine

### DISCUSSION THREAD

**Discussion by leuchtdiode** [pimcore/pimcore#7770] [General]
Hi,

can somebody please tell me why the PHP rendering engine will be removed in v7? Is there any specific reason?

Thx & BR,
Alex

**Comment by dvesh3** [comment]
Hi @leuchtdiode please follow the discussion here https://github.com/pimcore/pimcore/issues/7224

**Comment by leuchtdiode** [comment]
Ah, thank you ;)

**Comment by dpfaffenbauer** [comment]
TLDR: Symfony deprecated it, so Pimcore does too. If you still need it, it is available as Enterprise Extension

**Reply by leuchtdiode** [reply]
Thanks.

Can someone tell me how to use templating helpers written for PHP in twig though?

`CommonBundle\Templating\Helper\SiteInc:
public: true
tags:
      - { name: templating.helper, alias: siteInc }` (line breaks after present in YAML of course)

Using

`{{ siteInc(_context, '/snippets/header', [], true) }}`

instead of PHP version

`<?= $this->siteInc($this, '/snippets/header', [], true) ?>` (works well)

says: Unknown "siteInc" function.

BR,
Alex

**Reply by dvesh3** [reply]
@leuchtdiode write a Twig extension which calls your templating helper. something like https://github.com/pimcore/pimcore/blob/2662d44a8828ad8fc7c8d878020b115f8ba61d69/lib/Twig/Extension/SubrequestExtension.php#51

**Reply by leuchtdiode** [reply]
Thank you. I will give it a try :)

**Comment by leuchtdiode** [comment]
@dvesh3 Thank you, Twig extensions are just working fine 👍 

Another question: With PHP rendering engine I used headLink and headScript helper occassionally in my area views, so I could add CSS and JS on demand to my layout.

Starting with Twig this doesn't work any longer. It looks like although I call pimcore_head_link().appendStylesheet(...) and pimcore_head_script().appendFile(...) in my area, the correspnding HTML-tags are not being included in my head-section in the layout.

Any idea how I could accomplish this?

Thx & BR,
Alex

**Reply by dvesh3** [reply]
@leuchtdiode use deferred extension like this https://github.com/pimcore/demo/blob/ffc5837f555ebe6e024fafc1d8fe9d743d8ecd8d/app/Resources/views/layouts/layout.html.twig#L55

**Reply by leuchtdiode** [reply]
Works like a charm, thanks once again :)
